### PR TITLE
Add TypeScript Definitions

### PR DESCRIPTION
**Pull request by CarlosMayoralMX**
Hello, i added the TS definitions, i think that you just have to add them in the package.json declaration. 

A link to the reference 
https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

### CODE REVIEW

File: react-native-simple-dialogs.d.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { View, ViewProps,Text,Button,StyleProp,Modal,ModalProps,ButtonProps } from 'react-native';
```
Comment (ammichael) [inline react-native-simple-dialogs.d.ts]: `Modal` and `ModalProps` are not used

File: react-native-simple-dialogs.d.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    messageStyle?: StyleProp<Text>,
+    negativeButton?: ButtonProps<Button>,
+    positiveButton: ButtonProps<Button>
+    } 
+
+
+    class Dialog extends React.Component<DialogProps>{}
+    class ProgressDialog extends React.Component<ProgressDialogProps>{}
+    class ConfirmDialog  extends React.Component <ConfirmDialogProps>{}
+    class TouchableEffect extends React.Component <Props>{}
```
Comment (ammichael) [inline react-native-simple-dialogs.d.ts]: why dont export them directly and remove the last export line?

File: react-native-simple-dialogs.d.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+export {Dialog, ProgressDialog, ConfirmDialog,TouchableEffect }
```
Comment (ammichael) [inline react-native-simple-dialogs.d.ts]: My lint warned this wouldn't work wasn't being used:
image

Comment (CarlosMayoralMX) [reply react-native-simple-dialogs.d.ts]: Hello,

Every suggestion is well received. I'm not an expert in TS, so many things i code probably are improvable.

Saludos.

Comment (ammichael) [reply react-native-simple-dialogs.d.ts]: I'm not a TS expert myself, but this worked good to me:
```
  export class Dialog extends React.Component<DialogProps> {}
  export class ProgressDialog extends React.Component<ProgressDialogProps> {}
  export class ConfirmDialog extends React.Component<ConfirmDialogProps> {}
  export class TouchableEffect extends React.Component<Props> {}
```

Thanks!

**Comment by douglasjunior** [comment]
@ammichael @CarlosMayoralMX Thank you for your support guys.

I am also not an expert in "TS declaration files", so any help is welcome.

File: react-native-simple-dialogs.d.ts (hunk 1 above)
Comment (ammichael) [reply react-native-simple-dialogs.d.ts]: Also just noticed that these imports should be inside the declared module.

```
declare module 'react-native-simple-dialogs' {
  import * as React from 'react';
  import {
    View,
    ViewProps,
    Text,
    Button,
    StyleProp,
    ButtonProps,
  } from 'react-native';

type animationType = 'none' | 'slide' | 'fade';

{...}

```

**Comment by ammichael** [comment]
@CarlosMayoralMX the only changed really needed is importing the dependencies inside `declare module`. I created a [gist](https://gist.github.com/ammichael/7f9d794a6b1cbd0e8104df0b7008912a) with the changes I mentioned.

Could you please updated so @douglasjunior can merge it asap? (:

**Comment by douglasjunior** [comment]
Closing by #47, thank you guys.

**Pull request closed without merging**